lib/views/home_content/challenge/challenge_card.dart

```diff
@@ -0,0 +1,84 @@
+import "package:flutter/material.dart";
+import "package:flutter_svg/svg.dart";
+import "package:proxima/models/ui/challenge_card_data.dart";
+import "package:proxima/utils/ui/key_value_list_generator.dart";
+
+class ChallengeCard extends StatelessWidget {
+  static const challengeGroupIconKey = Key("challenge_group_icon");
+  static const challengeSingleIconKey = Key("challenge_single_icon");
+
+  static const _opacityWhenChallengedFinished = 0.3;
+  static const _challengeGroupAsset = "assets/icons/challenge_group.svg";
+  static const _challengeSingleAsset = "assets/icons/challenge_single.svg";
+
+  final ChallengeCardData challenge;
+
+  const ChallengeCard(this.challenge, {super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    KeyValueListGenerator listGenerator = KeyValueListGenerator(
+      style: DefaultTextStyle.of(context).style,
+    );
+
+    if (!challenge.isFinished) {
+      listGenerator.addPair("Distance to post", "${challenge.distance} meters");
+    } else {
+      listGenerator.addLine("Challenged finished!");
+    }
+    if (!challenge.isGroupChallenge) {
+      listGenerator.addPair("Time left", "${challenge.timeLeft} hours");
+    }
```

Comment:
Adding on to my previous comment, you would rather check if time left is not null here, and maybe show something like unknown if distance is null and the challenge is not finished.

Reply:
I'm not sure I agree, even after the rest of the refactor. Those values depend whether this is a `challenge` that is finished or not, and if it is a group one or not. We know those values can't be null by construction, and it makes sense `ChallengeCardData` is tightly linked to the `ChallengeCard` ui in my opinion.
What do you think?

Reply:
Good job making this clearer by adding constructors for each case.